app/controllers/devise_token_auth/concerns/set_user_by_token.rb

```diff
@@ -74,6 +74,10 @@ def update_auth_header
     @client_id = nil unless @used_auth_by_token
 
     if @used_auth_by_token and not DeviseTokenAuth.change_headers_on_each_request
+      # should not append auth header if @resource related token was
+      # cleared by sign out in the meantime
+      return if @resource.reload.tokens[@client_id].nil?
```

Comment:
@virginia-rodriguez - any implementation ideas that don't involve the additional `reload` call? i'm personally aiming for sub-200ms roundtrip responses in my apps these days and that's pretty hard to do with Rails and additional ActiveRecord calls.  =]